packages/api/src/services/actions/global.ts

```diff
@@ -1,71 +1,150 @@
 import assert from "assert";
 import { Json, Actions, AppState, CurrencySymbol, AllContractStates } from "../..";
-import * as Queries from "../../libs/queries";
 import bluebird from "bluebird";
 import { BigNumber } from "ethers";
 import { nowS } from "../../libs/utils";
+import { Handlers as EmpActions } from "./emp";
+import { Handlers as LspActions } from "./lsp";
 
 // actions use all the app state
 type Dependencies = AppState;
 type Config = undefined;
 
 export function Handlers(config: Config, appState: Dependencies): Actions {
   const { stats } = appState;
-  const queries = [Queries.Emp(appState), Queries.Lsp(appState)];
+  const contractActions = [EmpActions(config, appState), LspActions(config, appState)];
 
   const actions: Actions = {
     echo(...args: Json[]) {
       return args;
     },
     async listActive() {
       return bluebird.reduce(
-        queries,
-        async (result: AllContractStates[], query) => {
-          return result.concat(await query.listActive());
+        contractActions,
+        async (result: AllContractStates[], actions) => {
+          return result.concat((await actions.listActive()) as AllContractStates[]);
         },
         []
       );
     },
     async listExpired() {
       return bluebird.reduce(
-        queries,
-        async (result: AllContractStates[], query) => {
-          return result.concat(await query.listExpired());
+        contractActions,
+        async (result: AllContractStates[], actions) => {
+          return result.concat((await actions.listExpired()) as AllContractStates[]);
         },
         []
       );
     },
     // return the first address found
     async getState(address: string) {
       assert(address, "requires a contract address");
-      for (const query of queries) {
+      for (const action of contractActions) {
         try {
-          const state = await query.getAny(address);
-          return query.getFullState(state);
+          return await action.getState(address);
         } catch (err) {
           // do nothing
         }
       }
       throw new Error("Unable to find contract address " + address);
     },
-    async tvl(currency: CurrencySymbol = "usd") {
+    async globalTvl(currency: CurrencySymbol = "usd") {
       const sum = await bluebird.reduce(
-        queries,
-        async (sum, queries) => {
-          return sum.add((await queries.getTotalTvl(currency)) || "0");
+        contractActions,
+        async (sum, actions) => {
+          return sum.add(((await actions.tvl(undefined, currency)) as string) || "0");
         },
         BigNumber.from("0")
       );
       return sum.toString();
     },
-    async tvlHistoryBetween(start = 0, end: number = nowS(), currency: CurrencySymbol = "usd") {
+    async tvl(address: string, currency: CurrencySymbol = "usd") {
+      assert(address, "requires a contract address");
```

Comment:
these all folllow a very similar pattern, it loops through the contract type actions, and assumes if no error is thrown its a sucessful response, and keeps looping until it returns or runs out of types to check and throws an error

Reply:
Using a try-catch to weed out contract types that don't have this address seems a bit odd. Action also seems to be misnomer since each of these is just a contract type.

A structure like this would be preferable in my opinion:
```js
for (const contractType of contractTypes) {
  if (contractType.contracts.has(address)) {
    contractType.contracts.get(address).tvl();
  }
}
```

Things may not be architected this way, so that's more of an example than a concrete suggestion. Just saying that a more deliberate access mechanism makes 1) code more readable and 2) makes error detection and handling much better because we're not wrapping everything in a try-catch that uses errors as a proxy for non-existence.

I think any code that does the following should be avoided if at all possible:
```js
        } catch (err) {
          // do nothing
        }
```

Reply:
yea i think you are right, will update this

Reply:
👌